### PR TITLE
Completed Chapter3

### PR DESCRIPTION
**Pull request by adrianab**
### Solutions for Chapter 3

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter3.hs [hunk 1]
```diff
@@ -480,6 +517,60 @@ After defining the city, implement the following functions:
...
+
+sizeCity :: City -> Int
+sizeCity city = sum $ map sizeHouse $ cityHouse city
+
+sizeHouse :: House -> Int
+sizeHouse house = case house of
+      One -> 1
+      Two -> 2
+      Three -> 3
+      Four -> 4
```
Comment (vrom911) [inline src/Chapter3.hs:572]: Wonderful city 🌇 🏰

File: src/Chapter3.hs [hunk 2]
```diff
@@ -460,6 +487,16 @@ Create a simple enumeration for the meal types (e.g. breakfast). The one who
...
 
+data Meal
+      = Breakfast
+      | Lunch
+      | Brunch
+      | Dinner
+      | Supper
+      | Tea
+      | Elevenses
+      | Snack
```
Comment (vrom911) [inline src/Chapter3.hs:498]: Yummy 😋 ☕

File: src/Chapter3.hs [hunk 3]
```diff
@@ -560,22 +651,31 @@ introducing extra newtypes.
...
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Armor -> Dexterity -> Defense
+calculatePlayerDefense (Armor armor) (Dexterity dexterity) = Defense (armor * dexterity)
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Damage -> Defense -> Health -> Health
+calculatePlayerHit (Damage damage) (Defense defense) (Health health) = Health (health + defense - damage)
```
Comment (vrom911) [inline src/Chapter3.hs:678]: Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype

File: src/Chapter3.hs [hunk 4]
```diff
@@ -753,6 +853,15 @@ parametrise data types in places where values can be of any general type.
   maybe-treasure ;)
 -}
 
+newtype Dragon p = Dragon {power :: p}
+newtype TreasureChest t = Maybe t
```
Comment (vrom911) [inline src/Chapter3.hs:857]: Here you create a newtype with a constructor named `Maybe`. Though it is valid, I would not recommend that name, as `Maybe` is the name of the standard data type, so could create some confusion 🙂

File: src/Chapter3.hs [hunk 5]
```diff
@@ -971,6 +1095,30 @@ implement the following functions:
...
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Show, Eq, Bounded, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend x
+      | (x == Saturday) || (x == Sunday) = True
+      | otherwise = False
```
Comment (vrom911) [inline src/Chapter3.hs:1111]: Great!
Actually, you can inline that without using guards, as your checks already return the answer 🙂 

```suggestion
isWeekend x = x == Saturday || x == Sunday
```

File: src/Chapter3.hs [hunk 6]
```diff
@@ -971,6 +1095,30 @@ implement the following functions:
...
+
+nextDay :: WeekDay -> WeekDay
+nextDay x
+      | x == maxBound = minBound
+      | otherwise = succ x
+
+daysToParty :: WeekDay -> Int
+daysToParty x = mod (fromEnum Friday - fromEnum x) 7
```
Comment (vrom911) [inline src/Chapter3.hs:1119]: Perfect solution 👌🏼

**Review by vrom911 — APPROVED** [review]
Beautiful Chapter Three with nice data types 👷🏼 
Well done 👏🏼

**Comment by chshersh** [comment]
Thanks for participating in Learn4Haskell! Hope you enjoyed this Hacktoberfest 🥳 

I am going to close this PR as we can't merge the solutions into our course 🔐 

Even though the Hacktoberfest is over, we are going to continue reviewing, so feel free to open new PRs, if you haven't finished the course yet! 🎊

**Pull request closed without merging**